department-of-veterans-affairs/caseflow
# TimeSlot UI Fixes - Part of CASEFLOW-1115

[CASEFLOW-1115](https://vajira.max.gov/browse/CASEFLOW-1115) will be done across multiple PRs:
- Backend changes ([16147](https://github.com/department-of-veterans-affairs/caseflow/pull/16147)), adds db columns, falls back to existing behavior for number_of_slots and begins_at (slot_length_minutes has no existing behavior in db).
- Only show slots when scheduling virtual hearings ([16150](https://github.com/department-of-veterans-affairs/caseflow/pull/16150)).
- UI Fixes PR ([16149](https://github.com/department-of-veterans-affairs/caseflow/pull/16149)), stop using HearingTime to fix various bugs (temporary fix until we replace with custom time modal).
- Slot recalculation PR ([16161](https://github.com/department-of-veterans-affairs/caseflow/pull/16161)), branch is based on UI Fixes PR, accept new db values and calculate slots based on them.

### Description
This PR addresses the various UI bugs that are being caused by using the modalForms/HearingTime component to schedule custom times when TimeSlots are enabled.
- We will be creating a new component for this time selection in PI7.S4 with the work in [CASEFLOW-1070](https://vajira.max.gov/browse/CASEFLOW-1070)
- This work is intended to make CASEFLOW-1070 easier and allow testing of the work in this release.
- It's not code that we'll ever have users using in production.

### Testing Plan
Just a note about review/testing. It's going to be less worthwhile to review 'TimeSlot.test.js' on this PR since that file is changed quite a bit in [16161](https://github.com/department-of-veterans-affairs/caseflow/pull/16161).
1. Create a new hearing day for the oakland RO, virtual.
2. Go to schedule Veteran for Oakland RO, choose one of the veterans available for scheduling.
3. Schedule the veteran by clicking a time slot, and take a note of the time you scheduled them for, like this:
`H210411-687 8:45am PDT 11:45am EDT (slot)`
4. Go back to the Oakland RO and schedule another veteran, this time use the custom time dropdown. Make a note of the time like this
`H210411-697 1:30pm PDT 16:30pm EDT (dropown)`
5. Go back, and open the scheduling screen for one more veteran, don't schedule them, but verify:
    - The two hearings you scheduled show at the correct times in the slot listing.
    - If you toggle to the slot listing and back it doesn't change what slots are listed.
    - The slot listing shows correct slots (nothing before 8:30am or after 4:30pm local)
6. Verify that the appointments scheduled are correct on the daily docket. If you took notes as above, you can search for the docket number like `210411-697` to make it easier to verify the times.
